samcli/hook_packages/terraform/hooks/prepare/resources/apigw.py

```diff
@@ -91,3 +106,105 @@ class ApiGatewayStageProperties(ResourceProperties):
 
     def __init__(self):
         super(ApiGatewayStageProperties, self).__init__()
+
+
+def add_integrations_to_methods(cfn_dict: dict, gateway_integrations_cfn: Dict[str, List]) -> None:
+    """
+    Iterate through all the API Gateway methods in the translated CFN dict. For each API Gateway method,
+    search the internal integration resources using the integrations' unique identifier to find the
+    one that corresponds with that API Gateway method. Once found, append the properties of the internal
+    integration resource to match what CFN expects, which is an 'Integration' property on the API Gateway
+    method resource itself.
+
+    E.g.
+    AwsApiGatewayMethod:
+      Type: AWS::ApiGateway::Method
+      Properties:
+        Integration:
+          Uri: Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${function.Arn}/invocations
+          Type: AWS_PROXY
+
+    Parameters
+    ----------
+    cfn_dict: dict
+        Resultant CFN dict that will be mutated to append integrations to each API Gateway method
```

Comment:
we can send only the method CFN resources. It should be part of the ApiGateway method resources collected [here](https://github.com/aws/aws-sam-cli/blob/a070af6cb944b55fbd8f8da53846c54040fcb36e/samcli/hook_packages/terraform/hooks/prepare/resources/apigw.py#L69)